src/main/java/uk/gov/hmcts/ccd/domain/service/common/AttributeBasedAccessControlService.java

```diff
@@ -0,0 +1,278 @@
+
+    @Autowired
+    public AttributeBasedAccessControlService(final CompoundAccessControlService compoundAccessControlService,
+                                              final DefaultCaseDataAccessControl defaultCaseDataAccessControl,
+                                              final CaseDefinitionRepository caseDefinitionRepository) {
+        super(compoundAccessControlService);
+        this.defaultCaseDataAccessControl = defaultCaseDataAccessControl;
+        this.caseDefinitionRepository = caseDefinitionRepository;
+    }
+
+    @Override
+    public boolean canAccessCaseTypeWithCriteria(CaseTypeDefinition caseType,
+                                                 Set<String> userRoles,
+                                                 Predicate<AccessControlList> criteria) {
+        applyAccessProfileRules(caseType);
+
+        return super.canAccessCaseTypeWithCriteria(caseType,
+            userRoles,
+            criteria);
+    }
+
+    @Override
+    public boolean canAccessCaseStateWithCriteria(String caseState,
+                                                  CaseTypeDefinition caseType,
+                                                  Set<String> userRoles,
+                                                  Predicate<AccessControlList> criteria) {
+        applyAccessProfileRules(caseType);
+
+        return super.canAccessCaseStateWithCriteria(caseState,
+            caseType,
+            userRoles,
+            criteria);
+    }
+
+    @Override
+    public boolean canAccessCaseEventWithCriteria(String eventId,
+                                                  List<CaseEventDefinition> caseEventDefinitions,
+                                                  Set<String> userRoles,
+                                                  Predicate<AccessControlList> criteria) {
+        return super.canAccessCaseEventWithCriteria(eventId,
+            caseEventDefinitions,
+            userRoles,
+            criteria);
+    }
+
+    @Override
+    public boolean canAccessCaseFieldsWithCriteria(JsonNode caseFields,
+                                                   List<CaseFieldDefinition> caseFieldDefinitions,
+                                                   Set<String> userRoles,
+                                                   Predicate<AccessControlList> criteria) {
+        applyAccessProfileRules(caseFieldDefinitions);
+
+        return super.canAccessCaseFieldsWithCriteria(caseFields,
+            caseFieldDefinitions,
+            userRoles,
+            criteria);
+    }
+
+    @Override
+    public boolean canAccessCaseViewFieldWithCriteria(CommonField caseViewField,
+                                                      Set<String> userRoles,
+                                                      Predicate<AccessControlList> criteria) {
+        return super.canAccessCaseViewFieldWithCriteria(caseViewField,
+            userRoles,
+            criteria);
+    }
```

Comment:
We're not doing anything different here so no need to override the method?
```suggestion
```

Reply:
we are going to keep this change for now here as we need to update the code for the access-control logic